### ISSUE TITLE
[clang] ext_vector_type with an enum element type: Sema accepts it, ASTContext::getExtVectorType asserts

### ISSUE THREAD

**Issue by captainnemo9292** [llvm/llvm-project#225037]
https://godbolt.org/z/YnGejh663 (`-O0`)

```c
typedef enum { A } E;
typedef E V __attribute__((ext_vector_type(4)));
```
```
clang: clang/lib/AST/ASTContext.cpp:4802: clang::QualType clang::ASTContext::getExtVectorType(clang::QualType, unsigned int) const: Assertion `vecType->isBuiltinType() || vecType->isDependentType() || (vecType->isBitIntType() && llvm::isPowerOf2_32(vecType->castAs<BitIntType>()->getNumBits()))' failed.
 #10 clang::ASTContext::getExtVectorType(clang::QualType, unsigned int) const
 #11 clang::Sema::BuildExtVectorType(clang::QualType, clang::Expr*, clang::SourceLocation)
 #12 processTypeAttrs(...) SemaType.cpp
```

`BuildExtVectorType` checks the element with `isIntegerType()`, which is true for a complete enumeration type, but `getExtVectorType` requires a builtin (or `_BitInt`) element. `BuildVectorType` (GCC-style `vector_size`) rejects the same element type with `invalid vector element type 'E'`; the ext_vector_type path should do the same. No diagnostics are emitted before the assertion. Clang trunk, assertions build.
